sidecar/internal/service/reclaimspace.go

```diff
@@ -107,3 +139,38 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 		PostUsage: &proto.StorageConsumption{UsageBytes: res.PostUsage.UsageBytes},
 	}, nil
 }
+
+// k8sAccessModeToCSIAccessMode maps PersistentVolume access modes in Kubernetes
+// to CSI access mode.
+// Make sure to update this function as and when there are new modes introduced.
+func k8sAccessModeToCSIAccessMode(mode v1.PersistentVolumeAccessMode) csi.VolumeCapability_AccessMode_Mode {
+	switch mode {
+	case v1.ReadWriteOnce:
+		return csi.VolumeCapability_AccessMode_SINGLE_NODE_MULTI_WRITER
+	case v1.ReadOnlyMany:
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY
+	case v1.ReadWriteMany:
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER
+	case v1.ReadWriteOncePod:
+		return csi.VolumeCapability_AccessMode_SINGLE_NODE_SINGLE_WRITER
+	}
+
+	return csi.VolumeCapability_AccessMode_UNKNOWN
+}
+
+// getSecret get the secret details by name.
+func getSecret(kubeclient *kubernetes.Clientset, ns, name string) (map[string]string, error) {
```

Comment:
move this to new packages we need this for other controllers also like fencing

Reply:
okay